web-api/src/personalization/routes.rs

```diff
@@ -565,23 +549,25 @@ async fn personalize_knn_search_result(
             storage::Tag::get(storage, &id).await?,
         ),
         InputUser::Inline { history } => {
-            let config: &PersonalizationConfig = config.as_ref();
-            let history = trim_history(history, config.max_stateless_history_for_cois);
+            let history = trim_history(
+                history,
+                AsRef::<PersonalizationConfig>::as_ref(config).max_stateless_history_for_cois,
```

Comment:
:art: I do not thing that is stylisitcally or readability wise a good idea in general (same below).

I would argue `Trait::<generics>::method_with_self(value)` is for rust beginners somewhat of an advanced and confusing syntax feature and should be avoided (outside of  some exceptions where it makes things way simpler).

Reply:
it might not be very beginner friendly, but how exactly does this prevent us from using this syntax? there are no rust beginners working on the code and if that ever happens, then people can still ask questions.